File: packages/bachat_core/lib/src/models/udhaar.dart
class Udhaar {
  final String? id;
  final String type; // 'LENE' or 'DENE'
  final String personName;
  final double amount;
  final DateTime date;
  final String? notes;
  final bool isSettled;

  Udhaar({
    this.id,
    required this.type,
    required this.personName,
    required this.amount,
    required this.date,
    this.notes,
    this.isSettled = false,
  });

  factory Udhaar.fromJson(Map<String, dynamic> json) {
    return Udhaar(
      id: json['_id'],
      type: json['type'],
      personName: json['personName'],
      amount: (json['amount'] as num).toDouble(),
      date: DateTime.parse(json['date']),
      notes: json['notes'],
      isSettled: json['isSettled'] ?? false,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'type': type,
      'personName': personName,
      'amount': amount,
      'date': date.toIso8601String(),
      'notes': notes,
      'isSettled': isSettled,
    };
  }
}

File: backend/src/models/Udhaar.js
const mongoose = require('mongoose');

const udhaarSchema = mongoose.Schema({
    user: {
        type: mongoose.Schema.Types.ObjectId,
        required: true,
        ref: 'User'
    },
    type: {
        type: String,
        enum: ['LENE', 'DENE'], // LENE = To Receive, DENE = To Pay
        required: true
    },
    personName: {
        type: String,
        required: true
    },
    amount: {
        type: Number,
        required: true
    },
    date: {
        type: Date,
        default: Date.now
    },
    notes: {
        type: String
    },
    isSettled: {
        type: Boolean,
        default: false
    }
}, {
    timestamps: true
});

module.exports = mongoose.model('Udhaar', udhaarSchema);
